main.go

```diff
@@ -23,29 +25,89 @@ var (
 )
 
 func main() {
-	var err error
 	conf := readConfig()
 
-	lhs := parseFile(conf.Files.LHS)
-	rhs := parseFile(conf.Files.RHS)
+	if conf.JsonLines {
+		os.Exit(compareJsonLines(&conf))
+	} else {
+		lhs := parseFile(conf.Files.LHS)
+		rhs := parseFile(conf.Files.RHS)
+
+		os.Exit(compare(lhs, rhs, &conf))
+	}
+}
+
+func compareJsonLines(conf *config) int {
+	var err error
+	var lhs_cnt,rhs_cnt int
+	var lhs_file,rhs_file *os.File
+	var lhs_val,rhs_val interface{}
+
+	lhs_file, err = os.Open(conf.Files.LHS)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error: cannot read %s: %s\n", conf.Files.LHS, err.Error())
+		return statusReadError
+	}
+	defer lhs_file.Close()
+
+	rhs_file, err = os.Open(conf.Files.RHS)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error: cannot read %s: %s\n", conf.Files.RHS, err.Error())
+		return statusReadError
+	}
+	defer rhs_file.Close()
+
+	lhs_scanner := bufio.NewScanner(lhs_file)
+	rhs_scanner := bufio.NewScanner(rhs_file)
```

Comment:
Do you mean this https://godoc.org/encoding/json#Decoder.Decode? This is not the same because jsonStream(in the example) is array.

Reply:
`"encoding/json".Decoder.Decode` isn't limited to arrays, it can decode streams like the following one:

```json
{
    "a": 2,
    "b": 44
}
{
    "c": 23,
    "d": null
}
```

Reply:
Ok, i will check.

Let me explain my use case. 
I use jaydiff to test logstash filter works. Pipeline: input.json.log > logstash -> output.json.log. Result file(output.json.log) is comapred with expected result file(output_expected.json.log). One line is one json object. Log messages separated by the new line.

Reply:
JSON streams fit your use-case perfectly. Example of parsing a stream using `Decoder.Decode`: https://play.golang.org/p/Q2w5f4BBicn